02_bubble/jonathanheaden/main_test.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := strconv.Quote("[1 2 3 5]\n")
+	actual := strconv.Quote(buf.String())
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+
+func TestNegativesOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	fmt.Fprintln(out, bubble([]int{3, -2, 1, -5, 7}))
+
+	// Then
+	expected := strconv.Quote("[-5 -2 1 3 7]\n")
+	actual := strconv.Quote(buf.String())
+	r.Equalf(expected, actual, "Unexpected output in main()")
```

Comment:
When I was solving this lab I used `github.com/stretchr/testify/assert`, I noticed you have used `github.com/stretchr/testify/require` I just [read](https://godoc.org/github.com/stretchr/testify/require) up on the difference and it seems require will stop test execution on failure, is that what you want?

Reply:
For the purposes of this exercise I do want it to stop on fail, but thanks for pointing this out. Assert is what I would usually use